actors/builtin/reward/reward_state.go

```diff
@@ -103,20 +105,21 @@ func (st *State) withdrawReward(store adt.Store, owner addr.Address, currEpoch a
 	return withdrawableSum, nil
 }
 
+// AmountVested returns the `TokenAmount` value of funds vested in the reward at an epoch
 func (r *Reward) AmountVested(currEpoch abi.ChainEpoch) abi.TokenAmount {
-	elapsed := currEpoch - r.StartEpoch
 	switch r.VestingFunction {
 	case None:
 		return r.Value
 	case Linear:
-		vestDuration := big.Sub(big.NewInt(int64(r.EndEpoch)), big.NewInt(int64(r.StartEpoch)))
-		if big.NewInt(int64(elapsed)).GreaterThanEqual(vestDuration) {
+		elapsed := currEpoch - r.StartEpoch
+		vestDuration := r.EndEpoch - r.StartEpoch
+		if elapsed >= vestDuration {
 			return r.Value
 		}
 
 		// totalReward * elapsedEpoch / vestDuration
-		return big.Div(big.Mul(r.Value, big.NewInt(int64(elapsed))), vestDuration)
+		return big.Mul(r.Value, big.NewInt(int64(elapsed/vestDuration)))
 	default:
-		return abi.NewTokenAmount(0)
+		panic(fmt.Sprintf("invalid vesting function %v", r.VestingFunction))
```

Comment:
On balance I don't think we should panic here. A panic means "stop the blockchain" if such a message gets in a block somehow, and is quite difficult to deal with in validation. While I agree it certainly signals a programmer error somewhere if this happens, keeping funds locked while we arrange a protocol upgrade to fix the underlying issue is probably better than stopping.

Reply:
Oh, assumed since equivalent panics happen for similar checks that it would be okay, but I assume the difference is that you don't want this done on reward state for a call outside of the VM execution?

Reply:
They're judgement calls, and I'm not yet confident in them all yet. Some things would be so broken if violated that halting seems reasonable.